# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [2]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [3]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.4916416 0.7238413 0.6190031 0.4731950 0.6403095 0.6598940 0.5157355
2 0.3101377 0.1152227 0.4567100 0.6234007 0.0533412 0.1138845 0.1007758
  V8        V9        V10       V11        V12       V13       V14      
1 0.3673321 0.9944165 0.7362024 0.07439957 0.7337795 0.7221786 0.5291357
2 0.7948399 0.8365508 0.9984171 0.07183633 0.4351787 0.4223606 0.2134156
  V15       V16       V17        V18       V19        V20      
1 0.1327077 0.3179914 0.05428609 0.6478498 0.04049363 0.7743013
2 0.5123674 0.6348865 0.98556564 0.8964799 0.14355694 0.2918179

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>% gather()

  

head(df2)

key value    
1 V1  0.4916416
2 V1  0.3101377
3 V2  0.7238413
4 V2  0.1152227
5 V3  0.6190031
6 V3  0.4567100

## Assignment 2

We will generate some data:

In [13]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [14]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [17]:
df5 <- full_join(df3, df4, by="id")
df5


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook